app/controllers/robots_txt_controller.rb

```diff
@@ -46,12 +50,13 @@ def index
   # JSON that can be used by a script to create a robots.txt that works well with your
   # existing site.
   def builder
-    render json: fetch_robots_info
+    result = self.class.fetch_robots_info
+    overridden = SiteSetting.overridden_robots_txt
+    result[:overridden] = overridden if overridden.present?
+    render json: result
   end
 
-protected
-
-  def fetch_robots_info
+  def self.fetch_robots_info
```

Comment:
This method now does not return the robots info, it returns the default robots info (other code is supposed to know about this and return an override if it exists.) I would suggest renaming this to `fetch_default_robots_info`